# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: third_party/rust/naga/Makefile
# Commit: ef5dc3e04e5f
# Full Hash: ef5dc3e04e5f271eea0636ab3a495e95cc912f1d
# Author: Dzmitry Malyshau <dmalyshau@mozilla.com>
# Date: 2021-09-04 09:40:24
# Regressor Bug: 1726626
# File Overlap Count: 3
# Description:
#   Bug 1726626 - Move gfx/wgpu into a 3rd party dependency r=jgilbert,bholley
#   
#   This update makes wgpu a vendored dependency instead of having it in gfx/wgpu.
#   
#   ## Notes
# ==============================================================================

diff --git a/third_party/rust/naga/Makefile b/third_party/rust/naga/Makefile
--- a/third_party/rust/naga/Makefile
+++ b/third_party/rust/naga/Makefile
@@ -1,80 +1,91 @@
 .PHONY: all clean validate-spv validate-msl validate-glsl validate-dot validate-wgsl validate-hlsl
 .SECONDARY: boids.metal quad.metal
-SNAPSHOTS_IN=tests/in
-SNAPSHOTS_OUT=tests/out
+SNAPSHOTS_BASE_IN=tests/in
+SNAPSHOTS_BASE_OUT=tests/out
 
 all:
 	cargo fmt
 	cargo test --all-features --workspace
 	cargo clippy --all-features --workspace -- -D warnings
 
 clean:
 	rm *.metal *.air *.metallib *.vert *.frag *.comp *.spv
 
-%.metal: $(SNAPSHOTS_IN)/%.wgsl $(wildcard src/*.rs src/**/*.rs examples/*.rs)
+%.metal: $(SNAPSHOTS_BASE_IN)/%.wgsl $(wildcard src/*.rs src/**/*.rs examples/*.rs)
 	cargo run --features wgsl-in,msl-out -- $< $@
 
 %.air: %.metal
 	xcrun -sdk macosx metal -c $< -mmacosx-version-min=10.11
 
 %.metallib: %.air
 	xcrun -sdk macosx metallib $< -o $@
 
-%.dot: $(SNAPSHOTS_IN)/%.wgsl $(wildcard src/*.rs src/front/wgsl/*.rs src/back/dot/*.rs bin/naga.rs)
+%.dot: $(SNAPSHOTS_BASE_IN)/%.wgsl $(wildcard src/*.rs src/front/wgsl/*.rs src/back/dot/*.rs bin/naga.rs)
 	cargo run --features wgsl-in,dot-out -- $< $@
 
 %.png: %.dot
 	dot -Tpng $< -o $@
 
-validate-spv: $(SNAPSHOTS_OUT)/*.spvasm
+validate-spv: $(SNAPSHOTS_BASE_OUT)/spv/*.spvasm
 	@set -e && for file in $^ ; do \
-		echo "Validating" $${file#"$(SNAPSHOTS_OUT)/"};	\
+		echo "Validating" $${file#"$(SNAPSHOTS_BASE_OUT)/"};	\
 		cat $${file} | spirv-as --target-env vulkan1.0 -o - | spirv-val; \
 	done
 
-validate-msl: $(SNAPSHOTS_OUT)/*.msl
+validate-msl: $(SNAPSHOTS_BASE_OUT)/msl/*.msl
 	@set -e && for file in $^ ; do \
-		echo "Validating" $${file#"$(SNAPSHOTS_OUT)/"};	\
-		cat $${file} | xcrun -sdk macosx metal -mmacosx-version-min=10.11 -x metal - -o /dev/null; \
+		echo "Validating" $${file#"$(SNAPSHOTS_BASE_OUT)/"};	\
+		header=$$(head -n1 $${file});	\
+		cat $${file} | xcrun -sdk macosx metal -mmacosx-version-min=10.11 -std=macos-$${header:13:8} -x metal - -o /dev/null; \
 	done
 
-validate-glsl: $(SNAPSHOTS_OUT)/*.glsl
-	@set -e && for file in $(SNAPSHOTS_OUT)/*.Vertex.glsl ; do \
-		echo "Validating" $${file#"$(SNAPSHOTS_OUT)/"};\
+validate-glsl: $(SNAPSHOTS_BASE_OUT)/glsl/*.glsl
+	@set -e && for file in $(SNAPSHOTS_BASE_OUT)/glsl/*.Vertex.glsl ; do \
+		echo "Validating" $${file#"$(SNAPSHOTS_BASE_OUT)/"};\
 		cat $${file} | glslangValidator --stdin -S vert; \
 	done
-	@set -e && for file in $(SNAPSHOTS_OUT)/*.Fragment.glsl ; do \
-		echo "Validating" $${file#"$(SNAPSHOTS_OUT)/"};\
+	@set -e && for file in $(SNAPSHOTS_BASE_OUT)/glsl/*.Fragment.glsl ; do \
+		echo "Validating" $${file#"$(SNAPSHOTS_BASE_OUT)/"};\
 		cat $${file} | glslangValidator --stdin -S frag; \
 	done
-	@set -e && for file in $(SNAPSHOTS_OUT)/*.Compute.glsl ; do \
-		echo "Validating" $${file#"$(SNAPSHOTS_OUT)/"};\
+	@set -e && for file in $(SNAPSHOTS_BASE_OUT)/glsl/*.Compute.glsl ; do \
+		echo "Validating" $${file#"$(SNAPSHOTS_BASE_OUT)/"};\
 		cat $${file} | glslangValidator --stdin -S comp; \
 	done
 
-validate-dot: $(SNAPSHOTS_OUT)/*.dot
+validate-dot: $(SNAPSHOTS_BASE_OUT)/dot/*.dot
 	@set -e && for file in $^ ; do \
-		echo "Validating" $${file#"$(SNAPSHOTS_OUT)/"};	\
+		echo "Validating" $${file#"$(SNAPSHOTS_BASE_OUT)/"};	\
 		cat $${file} | dot -o /dev/null; \
 	done
 
-validate-wgsl: $(SNAPSHOTS_OUT)/*.wgsl
+validate-wgsl: $(SNAPSHOTS_BASE_OUT)/wgsl/*.wgsl
 	@set -e && for file in $^ ; do \
-		echo "Validating" $${file#"$(SNAPSHOTS_OUT)/"};	\
+		echo "Validating" $${file#"$(SNAPSHOTS_BASE_OUT)/"};	\
 		cargo run $${file}; \
 	done
 
-validate-hlsl: $(SNAPSHOTS_OUT)/*.hlsl
+validate-hlsl: SHELL:=/bin/bash # required because config files uses arrays
+validate-hlsl: $(SNAPSHOTS_BASE_OUT)/hlsl/*.hlsl
 	@set -e && for file in $^ ; do \
-		echo "Validating" $${file#"$(SNAPSHOTS_OUT)/"}; \
+		DXC_PARAMS="-Wno-parentheses-equality -Zi -Qembed_debug"; \
+		echo "Validating" $${file#"$(SNAPSHOTS_BASE_OUT)/"}; \
 		config="$$(dirname $${file})/$$(basename $${file}).config"; \
-		vertex=""\
-		fragment="" \
-		compute="" \
 		. $${config}; \
-		DXC_PARAMS="-Wno-parentheses-equality -Zi -Qembed_debug;" \
-		[ ! -z "$${vertex}" ] && echo "Vertex Stage:" && dxc $${file} -T $${vertex} -E $${vertex_name} $${DXC_PARAMS} > /dev/null; \
-		[ ! -z "$${fragment}" ] && echo "Fragment Stage:" && dxc $${file} -T $${fragment} -E $${fragment_name} $${DXC_PARAMS} > /dev/null; \
-		[ ! -z "$${compute}" ] && echo "Compute Stage:" && dxc $${file} -T $${compute} -E $${compute_name} $${DXC_PARAMS} > /dev/null; \
+		for (( i=0; i<$${#vertex[@]}; i++ )); do \
+			name=`echo $${vertex[i]} | cut -d \: -f 1`; \
+			profile=`echo $${vertex[i]} | cut -d \: -f 2`; \
+			(set -x; dxc $${file} -T $${profile} -E $${name} $${DXC_PARAMS} > /dev/null); \
+		done; \
+		for (( i=0; i<$${#fragment[@]}; i++ )); do \
+			name=`echo $${fragment[i]} | cut -d \: -f 1`; \
+			profile=`echo $${fragment[i]} | cut -d \: -f 2`; \
+			(set -x; dxc $${file} -T $${profile} -E $${name} $${DXC_PARAMS} > /dev/null); \
+		done; \
+		for (( i=0; i<$${#compute[@]}; i++ )); do \
+			name=`echo $${compute[i]} | cut -d \: -f 1`; \
+			profile=`echo $${compute[i]} | cut -d \: -f 2`; \
+			(set -x; dxc $${file} -T $${profile} -E $${name} $${DXC_PARAMS} > /dev/null); \
+		done; \
 		echo "======================"; \
 	done